ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js

```diff
@@ -105,25 +105,61 @@ const BaseFeeInput = () => {
   const [editingInGwei, setEditingInGwei] = useState(false);
 
   const [maxBaseFeeGWEI, setMaxBaseFeeGWEI] = useState(() => {
-    if (
-      estimateUsed !== PRIORITY_LEVELS.CUSTOM &&
-      advancedGasFeeValues?.maxBaseFee
+    if (estimateUsed !== PRIORITY_LEVELS.CUSTOM) {
+      if (advancedGasFeeValues?.maxBaseFeeGWEI) {
+        return advancedGasFeeValues.maxBaseFeeGWEI;
+      } else if (advancedGasFeeValues?.maxBaseFeeMultiplier) {
+        return multiplyCurrencyValues(
+          estimatedBaseFee,
+          advancedGasFeeValues.maxBaseFeeMultiplier,
+          numberOfDecimalsPrimary,
+        );
+      }
```

Comment:
If `estimateUsed !== PRIORITY_LEVELS.CUSTOM` is true we should use values from transaction and not `advancedGasFeeValues`.

Consider scenario when user has selected custom estimates and added his own values different from default. Now if user closes and open modal again above code will override his values.

Reply:
Should we save the updated values if the user hasn't clicked the `save` button? Because if we hit save the corresponding GWEI value will be saved, as it is in develop now.

Reply:
We should not update default setting if checkbox is not selected.

Reply:
I think I confused you, with the current changes, when the default gas settings are available we are setting the estimateUsed as CUSTOM which means we have a new scenario to use the advancedGasFeeValues when `estimateUsed === PRIORITY_LEVELS.CUSTOM`, and it is making it difficult to identify when we are using the advanced values and when the default values are updated, just by using the `estimateUsed !== PRIORITY_LEVELS.CUSTOM`.

This is the reason for the code change in this comment https://github.com/MetaMask/metamask-extension/pull/13099#discussion_r778009745